### PR TITLE
[Cart] Adding product to cart multiple times bug fix

### PR DESCRIPTION
**Pull request by Zales0123**
- added functionality of modify existing order item if adding product/variant has been added to cart before
- added some tests for new functionality
- added some missing tests for previous implementations that were not covered

### CODE REVIEW

File: spec/Modifier/OrderModifierSpec.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<?php
+
```
Comment (lchrusciel) [inline spec/Modifier/OrderModifierSpec.php]: We could go with strict types and void return type

File: tests/Controller/CartPutItemToCartApiTest.php [hunk 2]
```diff
@@ -545,6 +600,37 @@ public function it_adds_a_product_variant_based_on_option_to_the_cart()
...
+        $this->loadFixturesFromFile('shop.yml');
+
+        $token = 'SDAOSLEFNWU35H3QLI5325';
+
+        /** @var CommandBus $bus */
+        $bus = $this->get('tactician.commandbus');
+        $bus->handle(new PickupCart($token, 'WEB_GB'));
+
+        $data =
+            <<<EOT
```
Comment (lchrusciel) [inline tests/Controller/CartPutItemToCartApiTest.php]: Something went wrong with the formating

**Comment by Zales0123** [comment]
@lchrusciel fixed

**Comment by pjedrzejewski** [comment]
Nice!

**Pull request merged**